### PR TITLE
For debug reasons we now can store all incoming sources

### PR DESCRIPTION
**Pull request by annando**
See https://github.com/friendica/friendica/pull/11780#issuecomment-1305909455

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -1328,10 +1328,11 @@ public static function insert(array $item, int $notify = 0, bool $post_local = t
 			}
 		}
 
+		if (!empty($source) && ($transmit || DI::pConfig()->get($uid, 'debug', 'store_source'))) {
```
Comment (nupplaphil) [inline src/Model/Item.php]: Who can set this? It seems like a global config, so I'd rather say:

```suggestion
		if (!empty($source) && ($transmit || DI::config()->get('debug', 'store_source'))) {
```
Or am I wrong?

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Model/Item.php (hunk 1 above)
Comment (MrPetovan) [reply src/Model/Item.php]: You're right, and I have to take over this branch as well to fix the `Item\Source` module, this is only half the battle.

Comment (annando) [reply src/Model/Item.php]: Yeah, that pconfig thingy was a copy&paste issue.

File: src/Module/Moderation/Item/Source.php [hunk 2]
```diff
@@ -36,33 +36,46 @@ protected function content(array $request = []): string
...
+				'termslbl'    => $this->t('Terms'),
+				'taglbl'      => $this->t('Tag'),
+				'typelbl'     => $this->t('Type'),
+				'termlbl'     => $this->t('Term'),
+				'urllbl'      => $this->t('URL'),
+				'mentionlbl'  => $this->t('Mention'),
+				'implicitlbl' => $this->t('Implicit Mention'),
+				'error'       => $this->t('Error'),
+				'notfound'    => $this->t('Item not found'),
+				'nosource'    => $this->t('No source recorded, please make sure the <code>debug.store_source</code> config key is set in <code>config/local.config.php</code> for future items to have sources.'),
```
Comment (annando) [inline src/Module/Moderation/Item/Source.php]: I suggest to only display this message when the config isn't activated.

**Pull request merged**